### PR TITLE
add basic explore header

### PR DESCRIPTION
**Pull request by dsludwig**
Add a fixed header that links to other dask projects to improve discoverability.

screen shot 2018-09-06 at 11 09 17 am

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @TomAugspurger

**Comment by mrocklin** [comment]
@dsludwig thoughts on drop-down menus?  I suspect that in some cases we'll want to nest/group doc pages like the following:

-  Deploy
    - dask-yarn
    - dask-kubernetes
    - dask-jobqueue
-  Algorithms
    - dask arrays
    - dask dataframes
    - dask-ml
-   Other flat entry without nesting
-   Other nested element
    - a
    - b

**Comment by mrocklin** [comment]
Also, this looks great.  Thank you!

**Comment by TomAugspurger** [comment]
Thanks @dsludwig!

Do we want both the top and side navbars to be "sticky", so that they're always present? I tried to have the top navbar not be fixed position, but wasn't able to get the padding with the sidbar working.

**Comment by mrocklin** [comment]
> Do we want both the top and side navbars to be "sticky", so that they're always present? I tried to have the top navbar not be fixed position, but wasn't able to get the padding with the sidbar working.

Currently it appears to be sticky.  I think that's probably fine but don't have strong or informed opinions here.

**Comment by mrocklin** [comment]
> thoughts on drop-down menus? I suspect that in some cases we'll want to nest/group doc pages like the following

I was able to follow a [basic example](https://www.w3schools.com/howto/howto_css_dropdown_navbar.asp) to add a dropdown navbar. Is this the right approach or is there something cleaner?

**Comment by mrocklin** [comment]
I've pushed up an attempt at dropdowns within the navbar

image

**Comment by TomAugspurger** [comment]
For the record, I'm fine with both bars being sticky.

Agreed that dropdowns are the way to go. If you have them working with
plain CSS then lets go with that (though a slight style preference for
left-aligning the elements in the dropdown).

File: dask_sphinx_theme/layout.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{% block extrabody %}
+
+<nav id="explore-links">
+  <p class="caption"> Explore </p>
+
+
+<ul>
+  <li><a href="https://dask.pydata.org"> Home </a></li>
+  <li>
+    <a href="">Algorithms</a>
```
Comment (TomAugspurger) [inline dask_sphinx_theme/layout.html]: Placeholder, or do you think this is the correct name? I'm struggling to think of a better one, perhaps `Interfaces`?

**Comment by mrocklin** [comment]
image

**Comment by mrocklin** [comment]
@TomAugspurger I'm done futzing for a bit

**Comment by TomAugspurger** [comment]
Style:

- I see a small grayish box at the bottom of the sidebar. It seems to come from `wy-nav-side` having a `padding-bottom` of 2em. Do you see that? I'm not sure where it's coming from, but maybe we could explicitly override it.

**Comment by mrocklin** [comment]
Done

**Comment by TomAugspurger** [comment]
The only section of the navbar I'm unsure about is "Developer Docs". I don't see those as important enough to deserve a spot in the navbar.

In it's place I would maybe elevate "community", with links to the github org, the dask/dask community page,...

But, +1 to merge this now, and we'll address those later.

**Comment by TomAugspurger** [comment]
And we might want to think about putting the top and side navbars on different sides of the page, but that's for a followup as well.

**Pull request merged**